src/holon/rule_engine/repositories/repository_base.py

```diff
@@ -279,3 +287,27 @@ def discrete_attribute_passes_none_check(
         return True
 
     return getattr(object, attribute_name) != NONE_VALUE
+
+
+def instance_base_type(obj: object) -> type:
+    """
+    Get the base type of the model within our domain.
+    For example, for a DieselVehicleAsset return EnergyAsset,
+    and for a HouseGridConnection return GridConnection
+    """
+    return base_type(obj.__class__)
+
+
+def base_type(cls: type) -> type:
+    """
+    Get the base type of the model within our domain.
+    For example, for a DieselVehicleAsset return EnergyAsset,
+    and for a HouseGridConnection return GridConnection
+    """
+    while True:
+        base = cls.__base__
+        if base in (PolymorphicModel, Model, ClusterableModel, object):
+            return cls
+        else:
+            cls = base
```

Comment:
Mogen deze functienamen ge-prepend worden met `get_`? Dan is het duidelijker dat dit functies zijn die iets doen.

Reply:
Done